runtime/gc_base/FinalizerSupport.cpp

```diff
@@ -553,6 +553,17 @@ static int J9THREAD_PROC FinalizeSlaveThread(void *arg)
 
 		fns->internalEnterVMFromJNI(env);
 		
+#if defined(J9VM_GC_DYNAMIC_CLASS_UNLOADING)
+		if(slaveData->mode != FINALIZE_SLAVE_MODE_CL_UNLOAD)
+#endif /* J9VM_GC_DYNAMIC_CLASS_UNLOADING */
+		{
+			if (vm->processReferenceMonitor && (0 != finalizeListManager->getReferenceCount())) {
```

Comment:
Can you explicitly test for null? i.e. `if ((NULL != vm->processReferenceMonitor) && (0 != ....)`

Reply:
updated